### DISCUSSION TITLE
0/3 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 No preemption victims found for incoming pod

### DISCUSSION THREAD

**Discussion by nishit93-hub** [redhat-developer/odo#6983] [Q&A]
Hi,

I created a new Java project using the `odo init` command. After initializing new components, I ran the command `odo dev` However it is giving me the following error

`0/3 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 No preemption victims found for the incoming pod.`

I want to know do I need to create `PersistentVolumeClaims` first and define that in generated `devfile.yaml`

**Comment by rm3l** [comment]
Hi @nishit93-hub,

> I created a new Java project using the `odo init` command. After initializing new components, I ran the command `odo dev` However it is giving me the following error
> 
> `0/3 nodes are available: pod has unbound immediate PersistentVolumeClaims. preemption: 0/3 nodes are available: 3 No preemption victims found for the incoming pod.`
> 
> I want to know do I need to create `PersistentVolumeClaims` first and define that in generated `devfile.yaml`
>
You shouldn't need to create PVCs first. There are some PVCs that `odo` might request to sync the project sources and, if the devfile contains a `volume` component, `odo` would automatically create the PVCs for you (depending on whether the volume component is marked as `ephemeral` or not). See [How `odo` works](https://odo.dev/docs/development/architecture/how-odo-works#volumes) for more details.

To help investigate, can you please share your `devfile.yaml` file?
Also, can you tell us what type of cluster you are connected to? If possible, can you also share the output of the following command (`<component-name>` is the `metadata.name` field in the `devfile.yaml` file, and can also be extracted with `yq '.metadata.name' devfile.yaml`)?
```
kubectl describe pod --selector "app.kubernetes.io/instance=<component-name>"
```

**Reply by rm3l** [reply]
Also, as discussed with the team today, is this a temporary message that you are seeing? It might happen on some remote clusters, while the volumes are being provisioned, but this eventually succeeds and `odo dev` can start correctly.

**Comment by nishit93-hub** [comment]
Hi,

No this is a permanent issue. It tries to provision the volume again after some interval of time. I am attaching the screenshot


![Screenshot from 2023-07-20 11-58-49](https://github.com/redhat-developer/odo/assets/64061293/35e9be25-69a3-45b2-850e-6faeeff287a2)

**Reply by rm3l** [reply]
Ok, thanks for the information.

To help us investigate this, can you please share the complete output of `odo version`?
Also, can you check if you are already able to create a PVC properly and use it in a pod? For example, by running the following commands:
```shell
$ cat << EOF | kubectl apply -f - 
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: task-pv-claim
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
EOF

$ cat << EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:    
  name: mypod
spec:        
  containers:         
    - name: myfrontend
      image: nginx 
      volumeMounts:               
      - mountPath: "/var/www/html"
        name: mypd
  volumes:      
    - name: mypd            
      persistentVolumeClaim:
        claimName: task-pv-claim        
EOF

$ kubectl get po mypod -w
```

**Reply by nishit93-hub** [reply]
Hi,

Here is the Odo version.

```
⚠  unable to fetch the podman client version
odo v3.12.0 (c0127ce20)
```
Yes, I am able to create PVC and use it in a pod. I have other applications running on k8s with PVC attached

**Comment by dragongu** [comment]
nodes are available: pod has unbound immediate PersistentVolumeClaims
